### PR TITLE
[T5A2][T10-B4] Lim Khong Yew

### PR DESCRIPTION
**Pull request by khongyew**
Work in progress

### CODE REVIEW

**Comment by khongyew** [comment]
Ready for review

File: src/seedu/addressbook/commands/Command.java [hunk 1]
```diff
@@ -39,7 +39,9 @@ public static String getMessageForPersonListShownSummary(List<? extends ReadOnly
     /**
      * Executes the command and returns the result.
      */
-    public abstract CommandResult execute();
+    public abstract CommandResult execute() throws Exception;
+    
+    public abstract boolean isMutating();
```
Comment (mauris) [inline src/seedu/addressbook/commands/Command.java]: would be helpful if there's a comment to explain this method!

File: src/seedu/addressbook/logic/Logic.java [hunk 2]
```diff
@@ -85,7 +85,9 @@ public CommandResult execute(String userCommandText) throws Exception {
     private CommandResult execute(Command command) throws Exception {
         command.setData(addressBook, lastShownList);
         CommandResult result = command.execute();
-        storage.save(addressBook);
+        if(command.isMutating()){
+        	storage.save(addressBook);
```
Comment (mauris) [inline src/seedu/addressbook/logic/Logic.java]: coding standard violation, use 4-space instead of one-tab.

**Comment by mauris** [comment]
Some comments added. Feel free to add more comment if you have any questions. Once you are done, you may close the PR after reading comments.

**Pull request closed without merging**